### PR TITLE
feat(TDI-46430): Kafka Input Output Avro support

### PR DESCRIPTION
**Pull request by olhava**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-46430

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaInput/tKafkaInput_messages.properties [hunk 1]
```diff
@@ -55,3 +55,22 @@ KAFKA_CONSUMER_PROPERTIES.NAME=Kafka properties
 KAFKA_CONSUMER_PROPERTIES.ITEM.PROPERTY=Property
 KAFKA_CONSUMER_PROPERTIES.ITEM.VALUE=Value
 SAVE_OFFSET.NAME=Load the offset with the message.
+OUTPUT_TYPE.ITEM.CONSUMER_RECORD=ConsumerRecord
```
Comment (Tanya5091) [inline main/plugins/org.talend.designer.components.bigdata/components/tKafkaInput/tKafkaInput_messages.properties:58]: Does ConsumerRecord need to be written as one word?

**Review by Tanya5091 — DISMISSED** [review]

**Review by Tanya5091 — APPROVED** [review]
LGTM

**Comment by pyzhou-talend** [comment]
Leave a message here to keep tracking this PR. (I will receiver email after this comment) :)

**Review by Tanya5091 — DISMISSED** [review]
LGTM

**Review by ypiel-talend — APPROVED** [review]
LGTM

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaInput/0.10.0.x/tKafkaInput_begin_0.10.0.x.javajet [hunk 2]
```diff
@@ -66,53 +70,44 @@ for (java.util.Map.Entry<String, String> kafkaProperty : tKafkaInputUtil.getKafk
 <%
 }
 
-// SSL configuration
-if (tSetKeystoreUtil.useHTTPS()) {
```
[the remark this answers is not in the document]
Comment (olhava) [reply main/plugins/org.talend.designer.components.bigdata/components/tKafkaInput/0.10.0.x/tKafkaInput_begin_0.10.0.x.javajet:70]: Was moved to tKafkaInput_util.javajet

File: main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml [hunk 3]
```diff
@@ -183,7 +271,32 @@
...
-
+				
+			<!-- Kafka client libraries for 2.12-2.4.0 -->
+			<IMPORT MODULE_GROUP="KAFKA-CLIENT-LIB-2_12-2_4_0"
+				REQUIRED_IF="(KAFKA_VERSION=='KAFKA_2_4_0') AND (USE_EXISTING_CONNECTION=='false')" />
+
+			<!-- Schema registry, Avro serializer -->
+			<IMPORT NAME="avro-1.10.1" MODULE="avro-1.10.1.jar" MVN="mvn:org.apache.avro/avro/1.10.1"
+				REQUIRED_IF="(INPUT_TYPE == 'PRODUCER_RECORD') AND (USE_SCHEMA_REGISTRY == 'true') AND (SCHEMA_REGISTRY_KEY_SERIALIZER == 'AVRO' OR SCHEMA_REGISTRY_VALUE_SERIALIZER == 'AVRO')" />
+			<IMPORT NAME="jackson-core-2.9.5" MODULE="jackson-core-2.9.5.jar" MVN="mvn:com.fasterxml.jackson.core/jackson-core/2.9.5"
```
Comment (olhava) [inline main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml:282]: @pyzhou-talend could you please check if you need to align also the versions of jackson libs depending upon the used kafka version?
Seems like you've used the different versions for 2.4.x:
https://github.com/Talend/tbd-studio-se/pull/2342/files#diff-71837773f075bbbb7d14d37f1e731a8a92a1f6f241fbedb8127f0d8aa8a66249R365

Comment (pyzhou-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tKafkaOutput/tKafkaOutput_java.xml:282]: Yes. Better to align.

**Review by Tanya5091 — DISMISSED** [review]
LGTM

**Review by Tanya5091 — APPROVED** [review]
LGTM

**Pull request merged**